src/measures/finite.jl

```diff
@@ -871,6 +871,505 @@ end
+    multiclass_false_negative_rate(ŷ, y, class_w)
+
+False negative rate for multiclass observations `ŷ` and ground truth `y`.
+$DS_MICRO_MACRO `MulticlassFalseNegativeRate(; average=:micro)`. $CLASS_W
+
+For more information, run `info(multiclass_false_negative_rate)`.
+
+"""
+const multiclass_false_negative_rate = MulticlassFalseNegativeRate()
+const multiclass_falsenegative_rate  = MulticlassFalseNegativeRate()
+const multiclass_fnr                 = MulticlassFalseNegativeRate()
+const MFNR                           = MulticlassFalseNegativeRate
+const multiclass_miss_rate           = MFNR()
+
+"""
+    multiclass_false_discovery_rate
+
+$(docstring(MulticlassTruePositiveRate()))
+
+    multiclass_false_discovery_rate(ŷ, y)
+    multiclass_false_discovery_rate(ŷ, y, class_w)
+
+False discovery rate for multiclass observations `ŷ` and ground truth `y`.
+$DS_MICRO_MACRO `MulticlassFalseDiscoveryRate(; average=:micro)`. $CLASS_W
+
+For more information, run `info(multiclass_false_discovery_rate)`.
+
+"""
+const multiclass_false_discovery_rate = MulticlassFalseDiscoveryRate()
+const multiclass_falsediscovery_rate  = MulticlassFalseDiscoveryRate()
+const multiclass_fdr                  = MulticlassFalseDiscoveryRate()
+const MFDR                            = MulticlassFalseDiscoveryRate
+
+"""
+    multiclass_positive_predictive_value
+
+$(docstring(MulticlassPrecision()))
+
+    multiclass_positive_predictive_value(ŷ, y)
+    multiclass_positive_predictive_value(ŷ, y, class_w)
+
+Positive predictive value for multiclass observations `ŷ` and ground truth
+`y`. $DS_MICRO_MACRO `MulticlassPrecision(; average=:micro)`. $CLASS_W
+
+For more information, run `info(multiclass_positive_predictive_value)`.
+
+"""
+const multiclass_precision                 = MulticlassPrecision()
+const multiclass_ppv                       = MulticlassPrecision()
+const multiclass_positive_predictive_value = MulticlassPrecision()
+const multiclass_positivepredictive_value  = MulticlassPrecision()
+const MPPV                                 = MulticlassPrecision
+
+"""
+    multiclass_negative_predictive_value
+
+$(docstring(MulticlassNPV()))
+
+    multiclass_negative_predictive_value(ŷ, y)
+    multiclass_negative_predictive_value(ŷ, y, class_w)
+
+Negative predictive value for multiclass observations `ŷ` and ground truth
+`y`.$DS_MICRO_MACRO `MulticlassNPV(; average=:micro)`. $CLASS_W
+
+For more information, run `info(multiclass_negative_predictive_value)`.
+
+"""
+const multiclass_npv                       = MulticlassNPV()
+const multiclass_negative_predictive_value = MulticlassNPV()
+const multiclass_negativepredictive_value  = MulticlassNPV()
+const MNPV                                 = MulticlassNPV
+
+## INTERNAL FUCNTIONS ON MULTICLASS CONFUSION MATRIX
+
+_mtp(m::CM) = diag(m.mat)
+_mtn(m::CM)= sum(m.mat) .- (sum(m.mat, dims=2) .+ sum(m.mat, dims=1)' .+ diag(m.mat))
+_mfp(m::CM{N}) where N = N < 31 ? sum(m.mat, dims=2) .-= diag(m.mat) :
```

Comment:
To avoid error on julia v 1.0 try
```suggestion
_mfp(m::CM{N}) where N = (col_sums = sum(m.mat, dims=2); N < 31 ? col_sums.-= diag(m.mat) :
                                                                                                                  col_sums .- diag(m.mat) )    
```

Reply:
Yeah, I figured from logs; was trying to see if I could keep it 1 lined even in Julia 1.0 somehow. (Had to download it)

But, thank you for so quickly going through it and suggesting!

Reply:
@ven-k could you try bench-marking again with my suggestion [here](https://github.com/alan-turing-institute/MLJBase.jl/pull/430#discussion_r514157641) to see if we could drop the conditional statement?